doc/configuration.rst

```diff
@@ -1204,6 +1205,46 @@ further deferred, if desired).  The following produces only one plot::
   plt.plot([2, 2])
   plt.show()
 
+.. _dummy_images:
+
+Generating dummy images
+=======================
+
+For quick visualization of your gallery, especially during the writing process,
+Sphinx-Gallery allows you to build your gallery without executing any of the
+code (see :ref:`without_execution`). This however, can cause warnings about
+missing image files if you have manually written links to automatically
+generated images. To prevent these warnings you can tell Sphinx-Gallery to
+create a number of dummy images for an example.
+
+For example, you may have an example ('my_example.py') that generates 2 figures,
+which you then reference manually elsewhere, e.g.,:
+
+.. code-block:: rst
+
+    Below is a great figure:
+
+    .. figure:: ../auto_examples/images/sphx_glr_my_example_001.png
+
+    Here is another one:
+
+    .. figure:: ../auto_examples/images/sphx_glr_my_example_002.png
+
+To prevent missing image file warnings when building without executing, you
+can add the following to the example file::
+
+    # sphinx_gallery_dummy_images=2
+
+This will cause Sphinx-Gallery to generate 2 dummy images with the same
+naming convention and stored in the same location as images that would be
+generated when building with execution.
+
+.. note::
+    This configuration **only** works when ``plot_gallery`` is False. This
+    means that you will not need to remove any ``sphinx_gallery_dummy_images``
+    lines in your examples when you switch to building your gallery with
+    execution.
```

Comment:
This actually makes it slightly less useful. For example, a common use case in MNE-Python is to do:
```
PATTERN=plot_something make html_dev-pattern
```
which translates to a `sphinx-build .. -D sphinx_gallery_conf.filename_pattern=$PATTERN ...`. Under this condition, according to the documentation here, even with this `dummy_images` line in place, we would see warnings. In other words, I think it should place dummy images 1) if the example isn't going to be executed for any reason (i.e., in exclude, or not matching filename pattern, or plot_gallery=False), and 2) the images don't exist. This I think would make it a "always avoid a missing image warning" pattern, which seems more useful than a "avoid a missing image warning only when plot_gallery is False". But maybe I'm missing something

Reply:
Ah good point. It does behave as you suggest - if the example is not executed for any reason, the dummy images will be generated. I should update the doc.

With regard to 2 - good suggestion, will implement.